src/components/accessories/billsRecap/BillsRecap.tsx

```diff
@@ -0,0 +1,244 @@
+            Object.keys(summary.paymentsByMonthsOfYear)
+          ).sort((a, b) => monthList.indexOf(a) - monthList.indexOf(b))
+        : [];
+    return {
+      labels,
+      datasets: [
+        {
+          label: t("bill.payments"),
+          data:
+            labels.length != 0
+              ? labels.map((item) => summary.paymentsByMonthsOfYear[item] ?? 0)
+              : [],
+          borderColor: "rgb(255, 99, 132)",
+          backgroundColor: "rgba(255, 99, 132, 0.5)",
+        },
+        {
+          label: t("bill.debts"),
+          data:
+            labels.length != 0
+              ? labels.map((item) => summary.debtsByMonthsOfYear[item] ?? 0)
+              : [],
+          borderColor: "rgb(53, 162, 235)",
+          backgroundColor: "rgba(53, 162, 235, 0.5)",
+        },
+      ],
+    };
+  }, [summary]);
+
+  return (
+    <div className="bills__recap">
+      <div className="bills__recap__content">
+        <Doughnut
+          data={getDataFromTwoObjects("dailyRevenue", "dailyDebt")}
+          options={getOptions(
+            `${t("bill.today")} (${moment().format("DD/MM/YYYY")})`,
+            false
+          )}
+        />
+        <Pie
+          data={getDataFromTwoObjects("weeklyRevenue", "weeklyDebt")}
+          options={getOptions(
+            `${t("bill.week", {
+              start: moment().startOf("week").format("DD/MM/YYYY"),
+            })}`,
+            false
+          )}
+        />
+        <Pie
+          data={getDataFromTwoObjects("monthlyRevenue", "monthlyDebt")}
+          options={getOptions(
+            `${moment().format("MMMM")} ${moment().format("YYYY")}`,
+            false
+          )}
+        />
+        <Doughnut
+          data={getDataFromTwoObjects("annualRevenue", "annualDebt")}
+          options={getOptions(
+            `${t("bill.year")} ${moment().format("YYYY")}`,
+            false
+          )}
+        />
+        <Doughnut
+          data={getDataFromTwoObjects("currentUserCashIn", "currentUserDebt")}
+          options={getOptions(
+            t("bill.currentuser", { name: userCredentials?.displayName ?? "" }),
+            false
+          )}
+        />
+        <Bar
+          options={getOptions(t("bill.bestsellingbyquantity"))}
+          data={getDataFromObject(
+            "bestSellingByQuantity",
+            t("bill.totalquantity")
+          )}
+        />
+        <Bar
+          options={getOptions(t("bill.bestsellingbyoccurence"))}
+          data={getDataFromObject(
+            "bestSellingByOccurence",
+            t("bill.totaloccurence")
+          )}
+        />
+        <Bar
+          options={getOptions(t("bill.bestpatients"))}
+          data={getDataFromObject(
+            "bestPatientsByPayments",
+            t("bill.totalpayment")
+          )}
+        />
+        <Bar
+          options={getOptions(t("bill.mostindebtpatients"))}
+          data={getDataFromObject("mostIndebtedPatients", t("bill.totaldebt"))}
+        />
```

Comment:
Before this 4 charts, add a select list in order to select the year. Just for now put the select's values as static options like 2019/2020/2021/2022.

This value should be passed to the getDataFromObject in order to get only the data of that year.

Plus, for each label add the selected year.

Reply:
Ok I applied it. It is a good suggestion. Also I added an empty infox box placeholder when the selected data does'nt have  bills.